resources/todomvc/architecture-examples/preact/src/app/item.js

```diff
@@ -1,85 +1,49 @@
-import cx from 'classnames';
-import { h, Component } from 'preact';
-
-const ESCAPE_KEY = 27;
-const ENTER_KEY = 13;
-
-export default class TodoItem extends Component {
-    handleSubmit = () => {
-        let { onSave, onDestroy, todo } = this.props,
-            val = this.state.editText.trim();
+import cx from "classnames";
+// eslint-disable-next-line no-unused-vars
+import { h } from "preact";
+import { useState } from "preact/hooks";
+export default function TodoItem({ onSave, onRemove, onToggle, todo }) {
+    const [editing, setEditing] = useState(false);
+
+    function handleSubmit(e) {
+        const val = e.target.value.trim();
         if (val) {
             onSave(todo, val);
-            this.setState({ editText: val });
+            setEditing(false);
+        } else {
+            onRemove(todo);
         }
-        else {
-            onDestroy(todo);
-        }
-    };
+    }
 
-    handleEdit = () => {
-        let { onEdit, todo } = this.props;
-        onEdit(todo);
-        this.setState({ editText: todo.title });
-    };
+    function handleKeyDown(e) {
+        // prettier-ignore
+        if (e.key === "Escape" || e.key === "ESCAPE")
+            setEditing(false);
+        else if (e.key === "Enter" || e.key === "ENTER")
```

Comment:
same comment as above: can you please add a link to that bug number to fix the benchmark runner?

Reply:
merged main with the update